server/routerlicious/packages/routerlicious-base/src/riddler/api.ts

```diff
@@ -46,8 +49,19 @@ export function create(
 	router.post("/tenants/:id/validate", (request, response) => {
 		const tenantId = getParam(request.params, "id");
 		const includeDisabledTenant = getIncludeDisabledFlag(request);
-		const validP = manager.validateToken(tenantId, request.body.token, includeDisabledTenant);
-		handleResponse(validP, response);
+		const token = request.body.token;
+		const claims = decode(token) as ITokenClaims;
+		getGlobalTelemetryContext().bindProperties(
+			{ tenantId, documentId: claims.documentId },
+			() => {
+				const validP = manager.validateToken(
```

Comment:
If this is an async call, can this lead to those uncaught exceptions you've recently fixed? That is, do we need to add a .catch() to line 61?

Reply:
This was just a move, so I think it's out of scope for this PR, but that's a good point. The promise catch is handled within handleResponse, but yes this would fit the pattern we've been trying to remove. I'll follow up on this outside of this PR.